### PR TITLE
[#7589] Stickers packs horizontal scrolling

### LINKED ISSUE #7589: Stickers packs horizontal scrolling
The sticker pack menu, the one where you can select the sticker to send, should be horizontally page'able so you can swipe between sticker packs

### PR DESCRIPTION
**Pull request by flexsurfer**
fixes #7589

Image

### CODE REVIEW

File: src/status_im/ui/screens/chat/stickers/views.cljs [hunk 1]
```diff
@@ -27,33 +35,65 @@
...
+    (when pack-id
+      (re-frame/dispatch [:stickers/select-pack pack-id]))))
+
+(defview stickers-paging-panel [installed-packs selected-pack]
+  (letsubs [ref          (atom nil)
+            window-width [:dimensions/window-width]]
+    {:component-will-update (fn [_ [_ installed-packs selected-pack]]
+                              (update-scroll-position @ref installed-packs selected-pack window-width))
+     :component-did-mount   #(update-scroll-position @ref installed-packs selected-pack window-width)}
+    [react/scroll-view {:style                             {:flex 1} :horizontal true :paging-enabled true
```
Comment (dmitryn) [inline src/status_im/ui/screens/chat/stickers/views.cljs:79]: maybe move down `:horizontal` and `:paging-enabled` if they are not related to `:style`

**Review by dmitryn — APPROVED** [review]

File: src/status_im/ui/screens/chat/stickers/views.cljs [hunk 2]
```diff
@@ -27,33 +35,65 @@
...
 
-(defn- stickers-panel [stickers]
-  [react/scroll-view {:style {:flex 1} :condtent-container-style {:flex 1}}
-   [react/view {:style styles/stickers-panel}
-    (for [{:keys [uri] :as sticker} stickers]
-      ^{:key uri}
-      [react/touchable-highlight {:style    {:height 75 :width 75 :margin 5}
-                                  :on-press #(re-frame/dispatch [:chat/send-sticker sticker])}
-       [react/image {:style {:resize-mode :cover :width "100%" :height "100%"} :source {:uri uri}}]])]])
+(defview stickers-panel [stickers window-width]
```
Comment (yenda) [inline src/status_im/ui/screens/chat/stickers/views.cljs]: defview doesn't seem to be required here

File: src/status_im/ui/screens/chat/stickers/views.cljs [hunk 3]
```diff
@@ -27,33 +35,65 @@
...
+  [react/view {:width window-width :flex 1}
+   [react/scroll-view
+    [react/view {:style styles/stickers-panel}
+     (for [{:keys [uri] :as sticker} stickers]
+       ^{:key uri}
+       [react/touchable-highlight {:style    {:height 75 :width 75 :margin 5}
+                                   :on-press #(re-frame/dispatch [:chat/send-sticker sticker])}
+        [react/image {:style {:resize-mode :cover :width "100%" :height "100%"} :source {:uri uri}}]])]]])
```
Comment (yenda) [inline src/status_im/ui/screens/chat/stickers/views.cljs:46]: not related to this PR but is the uri a remote source? We might want to start using aggressive caching with https://github.com/DylanVann/react-native-fast-image if that is the case

Comment (flexsurfer) [reply src/status_im/ui/screens/chat/stickers/views.cljs:46]: yes that makes a lot of sense, thanks, i'll do it in separate PR

**Review by goranjovic — APPROVED** [review]

**Review by yenda — CHANGES_REQUESTED** [review]
remove unnecessary defview

**Comment by flexsurfer** [comment]
@yenda  done

File: src/status_im/utils/ethereum/stickers.cljs [hunk 4]
```diff
@@ -18,14 +18,14 @@
   (ethereum/call web3
                  (ethereum/call-params contract "getPackData(uint256)" (ethereum/int->hex pack-id))
                  (fn [_ data]
-                   (cb (abi-spec/decode (subs data 2) ["bytes4[]" "address" "bool" "uint256" "uint256" "bytes"])))))
+                   (cb (when-not (empty? (subs data 2)) (abi-spec/decode (subs data 2) ["bytes4[]" "address" "bool" "uint256" "uint256" "bytes"]))))))
```
Comment (yenda) [inline src/status_im/utils/ethereum/stickers.cljs]: is it intended that you call the cb function without arguments if the data is empty?
can you put (subs data 2) in a let binding and give it a name?

Comment (flexsurfer) [reply src/status_im/utils/ethereum/stickers.cljs]: yes it's intended

**Comment by flexsurfer** [comment]
@yenda done

**Review by yenda — APPROVED** [review]

**Comment by asemiankevich** [comment]
@jakubgs iOS build fails here, restart does not help

**Comment by asemiankevich** [comment]
@flexsurfer 

- i dont see stickers button on android

image

- on iOS i do see, but can't install

image

i believe it is the same as in nightly. Can you please check?

**Comment by asemiankevich** [comment]
alright i can now see stickers on iOS, my bad, i forgot to switch to ropsten. Android issue seems to me as separate one, it does not work in nightly too.

**Pull request merged**